src/script/lua_api/l_item.cpp

```diff
@@ -437,8 +437,32 @@ int LuaItemStack::l_peek_item(lua_State *L)
 int LuaItemStack::l_equals(lua_State *L)
 {
 	LuaItemStack *o1 = checkobject(L, 1);
-	ItemStack &item1 = o1->m_stack;
+
+	void *p = lua_touserdata(L, 2);
```

Comment:
I think `lua_type(L, 2) == LUA_TUSERDATA` should work as well
also there should be a (short) comment here explaining why it does this.

Reply:
seems reasonable, i based the code on `luaL_checkudata`. there's also apparently a `lua_isuserdata`, would that be better than your proposal?

Reply:
works too I guess